raft/log.go

```diff
@@ -1024,6 +1030,9 @@ func (l *Log) candidateLoop(closing <-chan struct{}) State {
 	elected := make(chan struct{}, 1)
 	go l.elect(term, elected, &wg)
 
+	electionTimer := l.Clock.ElectionTimer()
+	defer electionTimer.Stop()
```

Comment:
The candidate loop doesn't reset the timer so it can just defer when it exits. The follower loop needs to manually close it wherever it exits the function since it can get reset.